src/components/views/rooms/RoomHeader.tsx

```diff
@@ -14,37 +14,112 @@ See the License for the specific language governing permissions and
+import { Body as BodyText, IconButton } from "@vector-im/compound-web";
+import { Icon as VideoCallIcon } from "@vector-im/compound-design-tokens/icons/video-call.svg";
+import { Icon as VoiceCallIcon } from "@vector-im/compound-design-tokens/icons/voice-call.svg";
+import { Icon as ThreadsIcon } from "@vector-im/compound-design-tokens/icons/threads-solid.svg";
+import { Icon as NotificationsIcon } from "@vector-im/compound-design-tokens/icons/notifications-solid.svg";
+import { CallType } from "matrix-js-sdk/src/webrtc/call";
 
 import type { Room } from "matrix-js-sdk/src/matrix";
+import { _t } from "../../../languageHandler";
 import { useRoomName } from "../../../hooks/useRoomName";
 import DecoratedRoomAvatar from "../avatars/DecoratedRoomAvatar";
 import { RightPanelPhases } from "../../../stores/right-panel/RightPanelStorePhases";
 import RightPanelStore from "../../../stores/right-panel/RightPanelStore";
 import { useTopic } from "../../../hooks/room/useTopic";
 import { Flex } from "../../utils/Flex";
 import { Box } from "../../utils/Box";
+import { useRoomCallStatus } from "../../../hooks/room/useRoomCallStatus";
+import LegacyCallHandler from "../../../LegacyCallHandler";
+import defaultDispatcher from "../../../dispatcher/dispatcher";
+import { ViewRoomPayload } from "../../../dispatcher/payloads/ViewRoomPayload";
+import { Action } from "../../../dispatcher/actions";
+import { useRoomThreadNotifications } from "../../../hooks/room/useRoomThreadNotifications";
+import { NotificationColor } from "../../../stores/notifications/NotificationColor";
+import { useGlobalNotificationState } from "../../../hooks/useGlobalNotificationState";
+import SdkConfig from "../../../SdkConfig";
+import SettingsStore from "../../../settings/SettingsStore";
+
+/**
+ * A helper to transform a notification color to the what the Compound Icon Button
+ * expects
+ */
+function notificationColorToIndicator(color: NotificationColor): React.ComponentProps<typeof IconButton>["indicator"] {
+    if (color <= NotificationColor.None) {
+        return undefined;
+    } else if (color <= NotificationColor.Grey) {
+        return "default";
+    } else {
+        return "highlight";
+    }
+}
+
+/**
+ * A helper to show or hide the right panel
+ */
+function showOrHidePanel(phase: RightPanelPhases): void {
+    const rightPanel = RightPanelStore.instance;
+    rightPanel.isOpen && rightPanel.currentCard.phase === phase
+        ? rightPanel.togglePanel(null)
+        : rightPanel.setCard({ phase });
+}
 
 export default function RoomHeader({ room }: { room: Room }): JSX.Element {
     const roomName = useRoomName(room);
     const roomTopic = useTopic(room);
 
+    const { voiceCallDisabledReason, voiceCallType, videoCallDisabledReason, videoCallType } = useRoomCallStatus(room);
+    /**
+     * A special mode where only Element Call is used. In this case we want to
+     * hide the voice call button
+     */
+    const useElementCallExclusively = useMemo(() => {
```

Comment:
Would it make sense to use `useSettingValue` here? It would then update if the user changes the preferences.